x-pack/plugins/fleet/server/services/epm/agent/agent.ts

```diff
@@ -129,6 +130,23 @@ function toJsonHelper(value: any) {
 }
 handlebars.registerHelper('to_json', toJsonHelper);
 
+// toYamlHelper will convert non null object to a YAML string.
+function toYamlHelper(value: any) {
+  switch (typeof value) {
+    case 'string':
+      // Assume serialized yaml
+      return value;
+    case 'object':
+      // Null classified as object
+      if (isNil(value)) return;
+      const safeValue = replaceUndefinedObjectValues(value, true);
```

Comment:
Why do we need to replace `undefined` by null? I think there is still value to support undefined value if we want some keys to not be in the final yaml no?

Reply:
Passing objects with undefined key values causes `safeDump` to crash.
```
> yaml.safeDump({key: undefined})
Uncaught:
YAMLException: unacceptable kind of an object to dump [object Undefined]
    at writeNode (/Users/evgb/Documents/GitHub/kibana/node_modules/js-yaml/lib/js-yaml/dumper.js:779:13)
    at writeBlockMapping (/Users/evgb/Documents/GitHub/kibana/node_modules/js-yaml/lib/js-yaml/dumper.js:657:10)
    at writeNode (/Users/evgb/Documents/GitHub/kibana/node_modules/js-yaml/lib/js-yaml/dumper.js:750:9)
    at dump (/Users/evgb/Documents/GitHub/kibana/node_modules/js-yaml/lib/js-yaml/dumper.js:840:7)
    at Object.safeDump (/Users/evgb/Documents/GitHub/kibana/node_modules/js-yaml/lib/js-yaml/dumper.js:846:10) {
  reason: 'unacceptable kind of an object to dump [object Undefined]',
  mark: undefined
}
```